src/rdkafka.c

```diff
@@ -59,6 +60,14 @@
 #include <sys/timeb.h>
 #endif
 
+/* The below dependency is required for conversion of
+ * unsecured JWS compact serialization to base64URL.
+ * Not sure if this means we should include a
+ * WITH_SASL_OAUTHBEARER compile-time flag after all?
```

Comment:
that we should! change this #if to that.

But I suggest you move all OAUTHBEARER specifics to rdkafka_sasl_oauthbearer.c to have all oauth code in one place.
rdkafka_sasl_oauthbearer.h can be included without an #if-guard to make the code cleaner, but calls to oauth-stuff should be #if-guarded.

Reply:
👍 Created `rdkafka_sasl_oauthbearer.h` and do not #if-guard its inclusion as suggested, and all OAuth code is #if-guarded including the config parameters `sasl.oauthbearer.config` and `oauthbearer_token_refresh_cb`.  The appearance of OAUTHBEARER in the `sasl.mechanisms` parameter documentation is not #if-guarded since the SCRAM-related mechanisms are not, and it is probably best to keep it this way since there would be 4 combinations to deal with.